io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,394 @@
+    , signal_PointXYZRGBA (nullptr)
+    , fps_ (0)
+    , device_width_ (424)
+    , device_height_ (240)
+    , target_fps_ (30)
+    , file_name_or_serial_number_(file_name_or_serial_number)
+  {
+    signal_PointXYZ = createSignal<signal_librealsense_PointXYZ> ();
+    signal_PointXYZI = createSignal<signal_librealsense_PointXYZI> ();
+    signal_PointXYZRGB = createSignal<signal_librealsense_PointXYZRGB> ();
+    signal_PointXYZRGBA = createSignal<signal_librealsense_PointXYZRGBA> ();
+  }
+
+  RealSense2Grabber::~RealSense2Grabber ()
+  {
+    stop ();
+
+    disconnect_all_slots<signal_librealsense_PointXYZ> ();
+    disconnect_all_slots<signal_librealsense_PointXYZI> ();
+    disconnect_all_slots<signal_librealsense_PointXYZRGB> ();
+    disconnect_all_slots<signal_librealsense_PointXYZRGBA> ();
+
+    thread_.join ();
+
+    pipe_.stop ();
+  }
+
+  void 
+  RealSense2Grabber::start ()
+  {
+    if (isRunning ())
+      return;
+
+    running_ = true;
+
+    rs2::config cfg;
+
+    // capture from file
+    if (file_name_or_serial_number_.rfind (".bag") == file_name_or_serial_number_.length() - 4)
+    {
+      cfg.enable_device_from_file (file_name_or_serial_number_);
+    }
+    else
+    {
+      if (!file_name_or_serial_number_.empty ())
+        cfg.enable_device (file_name_or_serial_number_);
+
+      cfg.enable_stream (RS2_STREAM_COLOR, device_width_, device_height_, RS2_FORMAT_RGB8, target_fps_);
+      cfg.enable_stream (RS2_STREAM_DEPTH, device_width_, device_height_, RS2_FORMAT_Z16, target_fps_);
+      cfg.enable_stream (RS2_STREAM_INFRARED, device_width_, device_height_, RS2_FORMAT_Y8, target_fps_);
+    }    
+
+    rs2::pipeline_profile prof = pipe_.start (cfg);
+
+    assert (prof.get_stream (RS2_STREAM_COLOR).format () == RS2_FORMAT_RGB8);
+    assert (prof.get_stream (RS2_STREAM_DEPTH).format () == RS2_FORMAT_Z16);
+
+    ir_format_ = prof.get_stream (RS2_STREAM_INFRARED).format (); 
+
+    thread_ = std::thread (&RealSense2Grabber::threadFunction, this);
+
+  }
+
+  void 
+  RealSense2Grabber::stop ()
+  {
+    std::lock_guard<std::mutex> guard (mutex_);
+
+    quit_ = true;
+    running_ = false;
+  }
+
+  bool 
+  RealSense2Grabber::isRunning () const
+  {
+    std::lock_guard<std::mutex> guard (mutex_);
+
+    return running_;
+  }
+
+  float 
+  RealSense2Grabber::getFramesPerSecond () const
+  {
+    return fps_;
+  }
+
+  pcl::PointCloud<pcl::PointXYZ>::Ptr 
+  RealSense2Grabber::convertDepthToPointXYZ (const rs2::points & points)
+  {
+    pcl::PointCloud<pcl::PointXYZ>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZ> ());
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize (points.size ());
+
+    auto vertices_ptr = points.get_vertices ();
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
```

Comment:
`index` should be an unsigned type.

Reply:
Unfortunately I can't use unsigned ints for my for loop with OPENMP.  Looking at all the other parallel for loops in the project they all use int.

https://github.com/complexrehabsystems/pcl/blob/4e06bc0536536688fee99b0f0394064ef63420d3/keypoints/include/pcl/keypoints/impl/iss_3d.hpp#L303

Reply:
Note: The unsigned type loop index is available from OpenMP 3.0 (or later). OpenMP implemented in Visual C++ is 2.0.

Reply:
So is the PCL guideline to use `int` until OpenMP 3.0 is available in visual C++ ?

Reply:
There is no choice but to use signed type for support cross-platform (Visual C++) rather than guideline.
It is better to specify bits width according to the required size. I think it will be fine with 32 bit (<code>int32_t</code>).

I don't think Microsoft will actively support OpenMP 3.0 (and later) in Visual C++ future versions.
OpenMP 3.0 has been formulated in 10 years ago, but it has not been implemented in Visual C++ yet.